src/material-experimental/dialog/dialog-container.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ElementRef,
+  HostBinding,
+  ViewChild,
+  ComponentRef,
+  EmbeddedViewRef,
+  ChangeDetectorRef,
+  Component,
+  Optional,
+  Inject,
+  ViewEncapsulation,
+  ChangeDetectionStrategy,
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {trigger, state, style, transition, animate, AnimationEvent} from '@angular/animations';
+import {
+  BasePortalHost,
+  PortalHostDirective,
+  ComponentPortal,
+  TemplatePortal
+} from '@angular/cdk/portal';
+import {FocusTrapFactory} from '@angular/cdk/a11y';
+import {DialogConfig} from './dialog-config';
+import {Subject} from 'rxjs/Subject';
+
+
+export function throwDialogContentAlreadyAttachedError() {
+  throw Error('Attempting to attach dialog content after content is already attached');
+}
+
+
+/**
+ * Internal component that wraps user-provided dialog content.
+ * @docs-private
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'cdk-dialog-container',
+  template: `<ng-template cdkPortalOutlet></ng-template>`,
+  styles: [`
+    cdk-dialog-container {
+      background: white;
+      border-radius: 5px;
+      display: block;
+      padding: 10px;
+    }
+  `],
+  encapsulation: ViewEncapsulation.None,
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger('dialog', [
```

Comment:
IMO CDK components shouldn't include animations.

Reply:
Because we would like for the Cdk's Dialog to work as a base for other dialog services, we would like to establish both an example of how to do the animation as well as the structure for callbacks for the dialog animations.